perun-core/src/main/java/cz/metacentrum/perun/core/api/UsersManager.java

```diff
@@ -895,6 +895,36 @@ void changePassword(PerunSession sess, User user, String loginNamespace, String
 	void changeNonAuthzPassword(PerunSession sess, String i, String m, String password, String lang)
 		throws UserNotExistsException, LoginNotExistsException, PasswordChangeFailedException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException, PasswordStrengthException, PasswordResetLinkExpiredException, PasswordResetLinkNotValidException;
 
+	/**
+	 * Checks if the password reset request is valid. The request is valid, if it
+	 * was created, never used and hasn't expired yet.
+	 *
+	 * @param sess
+	 * @param token token for the request to check
+	 * @throws PasswordResetLinkExpiredException when the reset link expired
+	 * @throws PasswordResetLinkNotValidException when the reset link was already used or has never existed
+	 */
+	void checkPasswordResetRequestIsValid(PerunSession sess, String token) throws PasswordResetLinkExpiredException, PasswordResetLinkNotValidException;
```

Comment:
Please mark original methods as `@Deprecated`, since we will remove them in the next sprint.

Reply:
Done